src/libOpenImageIO-C/c-imageio.cpp

```diff
@@ -0,0 +1,257 @@
+// Copyright Contributors to the OpenImageIO project.
+// SPDX-License-Identifier: BSD-3-Clause
+// https://github.com/OpenImageIO/oiio
+
+
+#include <OpenImageIO/fmath.h>
+#include <OpenImageIO/imageio.h>
+
+#include "c-imageio_defines.h"
+#include "c-typedesc.h"
+
+using ImageInput  = OIIO::ImageInput;
+using ImageOutput = OIIO::ImageOutput;
+using ImageSpec   = OIIO::ImageSpec;
+using OIIO::bit_cast;
+
+extern "C" {
+
+stride_t OIIO_AutoStride = 0x8000000000000000L;
+
+
+
+ImageSpec*
+ImageSpec_new()
+{
+    return new ImageSpec;
+}
+
+
+
+void
+ImageSpec_delete(const ImageSpec* is)
+{
+    delete is;
+}
+
+
+
+ImageSpec*
+ImageSpec_new_with_dimensions(int xres, int yres, int nchans, TypeDesc fmt)
+{
+    return new ImageSpec(xres, yres, nchans,
+                         bit_cast<TypeDesc, OIIO::TypeDesc>(fmt));
+}
+
+
+
+ImageSpec*
+ImageSpec_copy(const ImageSpec* ii)
+{
+    return new ImageSpec(*ii);
+}
+
+
+
+void
+ImageSpec_attribute(ImageSpec* is, const char* name, TypeDesc fmt,
+                    const void* value)
+{
+    is->attribute(name, bit_cast<TypeDesc, OIIO::TypeDesc>(fmt), value);
+}
+
+
+
+int
+ImageSpec_width(const ImageSpec* is)
+{
+    return is->width;
+}
+
+
+
+int
+ImageSpec_height(const ImageSpec* is)
+{
+    return is->height;
+}
+
+
+
+int
+ImageSpec_nchannels(const ImageSpec* is)
+{
+    return is->nchannels;
+}
+
+
+
+const char*
+ImageSpec_channel_name(const ImageSpec* is, int chan)
+{
+    return is->channel_name(chan).c_str();
+}
+
+
+
+bool
+ImageSpec_getattribute(const ImageSpec* is, const char* name, TypeDesc type,
+                       void* value, bool casesensitive)
+{
+    return is->getattribute(name, bit_cast<TypeDesc, OIIO::TypeDesc>(type),
+                            value, casesensitive);
+}
+
+
+
+// FIXME: add Filesystem::IOProxy
+ImageInput*
+ImageInput_open(const char* filename, const ImageSpec* config, void* _ioproxy)
+{
+    auto ii = ImageInput::open(filename, config, nullptr);
+    return ii.release();
+}
+
+
+
+bool
+ImageInput_close(ImageInput* ii)
+{
+    return ii->close();
+}
+
+
+
+void
+ImageInput_delete(ImageInput* ii)
+{
+    delete ii;
+}
+
+
+
+const ImageSpec*
+ImageInput_spec(ImageInput* ii)
+{
+    return &ii->spec();
+}
+
+
+
+bool
+ImageInput_read_image(ImageInput* ii, int subimage, int miplevel, int chbegin,
```

Comment:
I assume this is just a first stab, and eventually we'll probably want somebody to go back and add read/write scanlines and tiles?

Reply:
Yes absolutely, I was just doing the bare minimum to get the test working. I'll hopefully get the rest of imageio.h bound over the weekend and submit a complete PR your Saturday night.